[Doc] GeoIP database service (#177)
logstash-plugins/logstash-filter-geoip: docs/index.asciidoc

If you would like to get Autonomous System Number(ASN) information, you can use the GeoLite2-ASN database.

[id="plugins-{type}s-{plugin}-database_license"]
==== Database License

> MaxMind changed from releasing the database under a CC license to a proprietary EULA,
> the EULA requires Logstash to update the MaxMind database for our users within 30 days of MaxMind updating their database.
> 
> For Logstash default distribution user, the plugin manages database for you.
> With the default setting, an auto-update makes sure database run in the latest version.
> For any reason Logstash fails to download database for 30 days, Logstash will stop the plugin pipeline in order to be compliant.
> If user config `database` option to a path, the plugin does not check for database update, and thus user has to be compliant on their own.
> 
> For Logstash open source user, the plugin use the Maxmind CC license database by default.

Reviewer: ```suggestion
https://www.maxmind.com[MaxMind] changed from releasing the GeoIP database under
a Creative Commons (CC) license to a proprietary end-user license agreement
(EULA).  The MaxMind EULA requires Logstash to update the MaxMind database
within 30 days of a database update. If Logstash fails to download the database
for 30 days, it will stop the pipeline in order to maintain compliance.

The GeoIP filter plugin can manage the database for users running the Logstash default
distribution, or you can manage
database updates on your own. The behavior is controlled by the `database`
setting. 
When you use the default `database` setting, the auto-update feature ensures that the plugin is
using the latest version of the database.
Otherwise, you are responsible for maintaining compliance.

The Logstash open source distribution uses the MaxMind Creative Commons license
database by default.
```

Reviewer: Overall: There are some instances of "Maxmind" in some of the older sections. Will you please change them all to  to "MaxMind" for accuracy?

Author: 👌

===== `database`

  * Value type is <<path,path>>
  * If not specified, the database defaults to the GeoLite2 City database that ships with Logstash.

The path to MaxMind's database file that Logstash should use. The default database is GeoLite2-City.
GeoLite2-City, GeoLite2-Country, GeoLite2-ASN are the free databases from MaxMind that are supported.
GeoIP2-City, GeoIP2-ISP, GeoIP2-Country are the commercial databases from MaxMind that are supported.

> Database auto-update applies to default distribution. When `database` point to user's database path,

Reviewer: ```suggestion
Database auto-update applies to default distribution. When `database` points to user's database path,
```

> auto-update will be disabled

Author: Be more explicit here to emphasize no auto-update if `database` is set

> auto-update will be disabled

Reviewer: ```suggestion
auto-update will be disabled.
```

> {logstash-ref}/plugins-filters-geoip.html#plugins-filters-geoip-database_license[see also database license]

Reviewer: ```suggestion
See
<<plugins-{type}s-{plugin}-database_license,Database License>> for more information.
```
